capabilities/host.go

```diff
@@ -74,60 +74,145 @@ func (h *Host) LookupHost(host string) ([]string, error) {
+	requestObj := sigstoreKeylessVerify{
+		Image:       image,
+		Keyless:     keyless,
+		Annotations: annotations,
+	}
+
+	return h.verify(requestObj)
+}
+
+// verify sigstore signatures of an image using keyless. Here, the provided
+// subject string is treated as a URL prefix, and sanitized to a valid URL on
+// itself by appending `/` to prevent typosquatting. Then, the provided subject
+// will satisfy the signature only if it is a prefix of the signature subject.
+// # Arguments
+// * `image` -  image to be verified
+// * `keyless`  -  list of issuers and subjects
+// * `annotations` - annotations that must have been provided by all signers when they signed the OCI artifact
+func (h *Host) VerifyKeylessPrefixMatchV2(image string, keylessPrefix []KeylessPrefixInfo, annotations map[string]string) (VerificationResponse, error) {
+	requestObj := sigstoreKeylessPrefixVerify{
+		Image:         image,
+		KeylessPrefix: keylessPrefix,
+		Annotations:   annotations,
+	}
+
+	return h.verify(requestObj)
+}
+
+// verify sigstore signatures of an image using keyless signatures made via
+// Github Actions.
+// # Arguments
+// * `image` -  image to be verified
+// * `owner` - owner of the repository. E.g: octocat
+// * `repo` - Optional. repo of the GH Action workflow that signed the artifact. E.g: example-repo. Optional.
+// * `annotations` - annotations that must have been provided by all signers when they signed the OCI artifact
+func (h *Host) VerifyKeylessGithubActionsV2(image string, owner string, repo string, annotations map[string]string) (VerificationResponse, error) {
+	requestObj := sigstoreGithubActionsVerify{
+		Image:       image,
+		Owner:       owner,
+		Repo:        repo,
+		Annotations: annotations,
+	}
+
+	return h.verify(requestObj)
+}
+
+// verify sigstore signatures of an image using a user provided certificate
+// # Arguments
+//   - `image` -  image to be verified
+//   - `certificate` - PEM encoded certificate used to verify the signature
+//   - `certificate_chain` - Optional. PEM encoded certificates used to verify `certificate`.
+//     When not specified, the certificate is assumed to be trusted
+//   - `require_rekor_bundle` - require the  signature layer to have a Rekor bundle.
+//     Having a Rekor bundle allows further checks to be performed,
+//     like ensuring the signature has been produced during the validity
+//     time frame of the certificate.
+//     It is recommended to set this value to `true` to have a more secure
+//     verification process.
+//   - `annotations` - annotations that must have been provided by all signers when they signed the OCI artifact
+func (h *Host) VerifyCertificateV2(image string, certificate string, certificateChain []string, requireRekorBundle bool, annotations map[string]string) (VerificationResponse, error) {
+	chain := make([][]rune, len(certificateChain))
+	for i, c := range certificateChain {
+		chain[i] = []rune(c)
+	}
```

Comment:
Hmm.... I tried, but this isn't what the Rust SDK does.
I'll look into it again.

Reply:
I think we cannot use just string in the payload. This is the program that I use to perform the serialization using Rust SDK to compare with what we are doing here:

```rust
use kubewarden_policy_sdk::host_capabilities::SigstoreVerificationInputV2;

fn main() {
    let certificate = "Testing certificate";
    let certificate_chain = Some(vec!["certificate1".to_string(), "certificate2".to_string()]);
    let require_rekor_bundle = false;
    let annotations = None;

    let chain: Option<Vec<Vec<u8>>> =
        certificate_chain.map(|c| c.iter().map(|cert| cert.as_bytes().to_vec()).collect());

    let input = SigstoreVerificationInputV2::SigstoreCertificateVerify {
        image: "myimage:latest".to_string(),
        certificate: certificate.as_bytes().to_vec(),
        certificate_chain: chain,
        require_rekor_bundle,
        annotations,
    };

    // Serialize it to a JSON string.
    let j3 = serde_json::to_string(&input).unwrap();

    // Print, write to a file, or send to an HTTP server.
    println!("{}", j3);

}
```

This is the output of the program:
```shell
make build run
K8S_OPENAPI_ENABLED_VERSION=1.24 cargo build --release
    Finished release [optimized] target(s) in 0.06s
K8S_OPENAPI_ENABLED_VERSION=1.24 cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/testpkg`
{"type":"SigstoreCertificateVerify","image":"myimage:latest","certificate":[84,101,115,116,105,110,103,32,99,101,114,116,105,102,105,99,97,116,101],"certificate_chain":[[99,101,114,116,105,102,105,99,97,116,101,49],[99,101,114,116,105,102,105,99,97,116,101,50]],"require_rekor_bundle":false,"annotations":null}
```

If we take a look in the payload [definition](https://github.com/kubewarden/policy-sdk-rust/blob/d7cba0ae45762c0b746b799a903f2c84845abe11/src/host_capabilities/mod.rs#L97-L101) we can see that the certificates are send over the wire as bytes. Not strings. Therefore, it's necessary to convert the string into bytes. I would say that we may keep the object using string and just convert it to bytes in the serialization step. But we cannot just JSON strings. It will not work. 

Is there a easyjson config to do that for us automatically?

Besides that, this show us another docs issue that needs to by fixed.